### PR TITLE
WIP - Refactorize api elements

### PR DESCRIPTION
**Pull request by 7eben**

### CODE REVIEW

**Review by jrpedrianes — CHANGES_REQUESTED** [review]

File: src/main/java/com/kuflow/rest/client/resource/ElementValueWrapperResource.java [hunk 1]
```diff
@@ -52,66 +52,66 @@ Object getRawValue() {
...
         return toElementValuesResource(value);
     }
 
-    public static ElementValuesResource of(String... values) {
+    public static ElementValueWrapperResource of(String... values) {
         return toElementValuesResource(CastUtils.cast(values));
     }
 
-    public static ElementValuesResource of(Integer value) {
+    public static ElementValueWrapperResource of(Integer value) {
```
[the remark this answers is not in the document]
Comment (7eben) [reply src/main/java/com/kuflow/rest/client/resource/ElementValueWrapperResource.java:81]: For now, I left Double and Integers... but I do not know. Maybe I remove them to not implement them internally (mappers, jsondata) in Kuflow in this iteration.

[the remark this answers is not in the document]
Comment (jrpedrianes) [reply src/main/java/com/kuflow/rest/client/resource/ElementValueWrapperResource.java:81]: ok, only remember to remove these if, in the end, we don't give support to this in the API.

**Review by jrpedrianes — APPROVED** [review]

File: src/main/java/com/kuflow/rest/client/resource/ProcessElementValueItemResource.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        if (this.value == null) {
+            return null;
+        }
+
+        if (this.value instanceof Number) {
+            return ((Number) this.value).doubleValue();
+        }
+
+        return Double.valueOf(this.value.toString());
+    }
```
Comment (jrpedrianes) [inline src/main/java/com/kuflow/rest/client/resource/ProcessElementValueItemResource.java:56]: I think that we need to add another 'getValueAsXXX' method for dates. Now we can have the following field types: Text, Number or Date, ie:

getValueAsInstant ¿?

File: src/main/java/com/kuflow/rest/client/resource/ProcessElementValueWrapperResource.java [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+    }
+
+    public static ProcessElementValueWrapperResource of(String value) {
+        return toProcessElementWrapperResource(value);
+    }
+
+    public static ProcessElementValueWrapperResource of(Double value) {
+        return toProcessElementWrapperResource(value);
+    }
+
```
Comment (jrpedrianes) [inline src/main/java/com/kuflow/rest/client/resource/ProcessElementValueWrapperResource.java:38]:
```suggestion

    public static ProcessElementValueWrapperResource of(Instant value) {
        return toProcessElementWrapperResource(value);
    }
    
```

File: src/main/java/com/kuflow/rest/client/resource/ProcessElementValueItemResource.java (hunk 2 above)
Comment (7eben) [reply src/main/java/com/kuflow/rest/client/resource/ProcessElementValueItemResource.java:56]: Is only getValueAsDate() -> LocalDate
and getValuesAsDate()-> List<LocalDate>
?

Comment (jrpedrianes) [reply src/main/java/com/kuflow/rest/client/resource/ProcessElementValueItemResource.java:56]: Are we storing the TimeZone?

If not, you are right that we must use LocalDate.

Maybe I think that is better to use `getValueAsLocalDate`.

Also, we need to add all the `X.of(LocalDate value)`, we need to add the methods for both sides, to read and create.

Comment (7eben) [reply src/main/java/com/kuflow/rest/client/resource/ProcessElementValueItemResource.java:56]: An relataded methos and test

Comment (jrpedrianes) [reply src/main/java/com/kuflow/rest/client/resource/ProcessElementValueItemResource.java:56]: yes, all that stuffs

**Review by jrpedrianes — APPROVED** [review]
👏 💪

**Pull request merged**